backend/service_objects.md

```diff
@@ -0,0 +1,144 @@
+# Intro to Service Objects:
+The Model View Controller (MVC) architecture that Rails provides is quite useful for organizing and separating code by functionality, and service objects are another step down the path of the “place for everything and everything in its place” mentality.
+
+If models are supposed to contain its own behaviors and controllers are meant to route incoming requests and views take and display data, then where do you put the other “stuff”? Often the answer is service objects. In particular, service objects are useful when a particular “job” involves multiple models, several steps, calls to outside APIs, or otherwise doesn’t fit neatly into any of the other MVC buckets.
+
+Say we are building a backend to facilitate signing up board game players. Among other things, our API will need to confirm that a player has signed up for a game. At the very least, this would involve both the `Player` and `Game` models. And while we could put all of this “business logic” into the controller that responds to this request, this could quickly pollute our controllers.
+
+Instead, we might create a service object that will bundle all the tasks related to this behavior. The controller will be responsible for calling the service object, and the service object will be responsible for executing the job. This way, our controllers are simply receiving traffic and signaling that certain jobs are needed; and, our service objects will be self-contained, of a singular purpose, and reusable - all good things.
+
+Let’s assume we have User, Player, and Game models, where:
+* a User has many Players
+* a Player has one User
+* a Player has one Game
+* Games have many Players
+
+Here is how a service object that confirms a player is playing in a game might look:
+/(below we’ll break down the individual components)/
+
+```
```

Comment:
You can put ` ```ruby` at the top of a code sample and it will format it some
I.E.
```ruby
  def initialize(game_id:, players_user_ids:)
    @game_id = game_id
    @players_user_ids = players_user_ids
  end
```